tests/functional/test_kinesis.py

```diff
@@ -0,0 +1,92 @@
+# language governing permissions and limitations under the License.
+import json
+import time
+from base64 import b64decode
+from uuid import uuid4
+from tests import unittest, BaseSessionTest, ClientHTTPStubber
+
+
+class TestKinesisListStreams(BaseSessionTest):
+    def setUp(self):
+        super(TestKinesisListStreams, self).setUp()
+        self.stream_name = "kinesis-test-stream"
+        self.region = "us-east-1"
+        self.client = self.session.create_client("kinesis", self.region)
+        self.http_stubber = ClientHTTPStubber(self.client)
+        self.http_stubber.add_response()
+
+    def test_can_put_stream_blob(self):
+        unique_data = str(uuid4())
+        with self.http_stubber as stub:
+            self.client.put_record(
+                StreamName=self.stream_name, PartitionKey="foo", Data=unique_data
+            )
+            self.assertEqual(len(stub.requests), 1)
+            # TODO: Decode for 3.4 and 3.5 (remove after deprecation)
+            request = json.loads(stub.requests[0].body.decode("utf-8"))
+            self.assertEqual(request["StreamName"], self.stream_name)
+            self.assertEqual(request["PartitionKey"], "foo")
+            self.assertEqual(
+                b64decode(request["Data"]).decode("utf-8"),
+                unique_data
+            )
+
+    def test_can_put_records_single_blob(self):
+        unique_data = str(uuid4())
+        with self.http_stubber as stub:
+            self.client.put_records(
+                StreamName=self.stream_name,
+                Records=[{"Data": unique_data, "PartitionKey": "foo"}],
+            )
+            self.assertEqual(len(stub.requests), 1)
+            # TODO: Decode for 3.4 and 3.5 (remove after deprecation)
+            request = json.loads(stub.requests[0].body.decode("utf-8"))
+            self.assertEqual(len(request["Records"]), 1)
+            self.assertEqual(request["StreamName"], self.stream_name)
+
+            record = request["Records"][0]
+            self.assertEqual(record["PartitionKey"], "foo")
+            self.assertEqual(
+                b64decode(record["Data"]).decode("utf-8"),
+                unique_data
+            )
+
+    def test_can_put_records_multiple_blob(self):
+        with self.http_stubber as stub:
+            self.client.put_records(
+                StreamName=self.stream_name,
+                Records=[
+                    {"Data": "foobar", "PartitionKey": "foo"},
+                    {"Data": "barfoo", "PartitionKey": "foo"},
+                ],
+            )
+            self.assertEqual(len(stub.requests), 1)
+            # TODO: Decode for 3.4 and 3.5 (remove after deprecation)
+            request = json.loads(stub.requests[0].body.decode("utf-8"))
+            self.assertEqual(len(request["Records"]), 2)
+
+            record_foobar = request["Records"][0]
+            record_barfoo = request["Records"][1]
+            self.assertEqual(
```

Comment:
It might be worth adding a helper assertion method like `assert_base64encoded_record`, maybe we can come up with a better name, to help make the test case a little easier to read and also so we do not have duplicate the same logic for each assertion